### PR TITLE
Add gherkin syntax highlighting

### PR DESCRIPTION
**Pull request by SamyAB**
Add tree-sitter grammar and highlights for the [gherkin language](https://cucumber.io/docs/gherkin/reference/) aka .feature files.

This is what it looks like with the `dracula` theme:
gherkin_in_helix

This was requested in https://github.com/helix-editor/helix/issues/3115 but it does not completely solve the issue as this PR does not add the LSP support.

### CODE REVIEW

**Comment by the-mikedavis** [comment]
Could you add a license to tree-sitter-gherkin? That will make packaging easier for downstream distributions like Fedora

**Comment by SamyAB** [comment]
@the-mikedavis I'll do it this evening!

**Comment by SamyAB** [comment]
@the-mikedavis license added, and `languages.toml` updated to make the `source.rev` of the grammar point to the commit with the license

File: languages.toml [hunk 1]
```diff
@@ -3680,6 +3680,19 @@ language-servers = [
...
 grammar = "javascript"
 
+[[language]]
+name = "gherkin"
+scope = "source.feature"
+file-types = ["feature"]
+comment-token = "#"
+indent = { tab-width = 2, unit = "  " }
+grammar = "gherkin"
+roots = []
```
Comment (the-mikedavis) [inline languages.toml:3690]:
```suggestion
```

`roots` is defaulted to `[]` so we can remove this line

Comment (SamyAB) [reply languages.toml]: Removed

File: languages.toml [hunk 2]
```diff
@@ -3680,6 +3680,18 @@ language-servers = [
 indent = { tab-width = 2, unit = "  " }
 grammar = "javascript"
 
+[[language]]
+name = "gherkin"
+scope = "source.feature"
+file-types = ["feature"]
+comment-token = "#"
+indent = { tab-width = 2, unit = "  " }
+grammar = "gherkin"
```
Comment (the-mikedavis) [inline languages.toml:3689]: one more small thing - this is also the default so it can be removed too

```suggestion
```

Comment (SamyAB) [reply languages.toml]: Removed

**Review by the-mikedavis — APPROVED** [review]

File: languages.toml [hunk 3]
```diff
@@ -3680,6 +3680,17 @@ language-servers = [
...
+name = "gherkin"
+scope = "source.feature"
+file-types = ["feature"]
+comment-token = "#"
+indent = { tab-width = 2, unit = "  " }
+
+[[grammar]]
+name = "gherkin"
+source = { git = "https://github.com/SamyAB/tree-sitter-gherkin", rev = "43873ee8de16476635b48d52c46f5b6407cb5c09" }
+
```
Comment (the-mikedavis) [inline languages.toml:3693]: ok actually one last thing 😅

Could you move this block above `gjs` or below `gts`? Where it is now it interrupts `gjs`'s block and accidentally takes its `auto-pairs` config

Comment (SamyAB) [reply languages.toml]: I did not see that it was in the middle of a block, my bad :sweat_smile:

It's now bellow `gts`

**Review by the-mikedavis — APPROVED** [review]

**Comment by archseer** [comment]
Can you rebase the PR? Looks like the CI was broken at that point in time

**Comment by SamyAB** [comment]
@archseer rebase done

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**